sys/include/bitfield.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2015 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_bitfield bitfields of arbitrary length
+ * @ingroup     sys
+ * @brief       bitfields of arbitrary length
+ * @{
+ *
+ * @brief       bitfields operations on bitfields of arbitrary length
+ *
+ * @note        Code taken mostly from
+ *              <a href="http://stackoverflow.com/questions/1590893/error-trying-to-define-a-1-024-bit-128-byte-bit-field">
+ *              Stackoverflow, User Christoph</a>
+ *
+ * @author      Oliver Hahm <oliver.hahm@inria.fr>
+ */
+
+#ifndef BITFIELD_H
+#define BITFIELD_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Declare a bitfield of a given size
+ *
+ * @note    SIZE should be a constant expression. This avoids variable length
+ *          arrays and problems resulting from being evaluated twice
+ */
+#define BITFIELD(NAME, SIZE) \
+     uint8_t NAME[(SIZE) / 8 + ((SIZE) % 8 != 0)]
```

Comment:
another way would be to write it like:
```c
uint8_t NAME[(SIZE + 7) / 8]
```
I am unsure which one is more readable.